x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/saml/sp/ServiceProviderDefaults.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.idp.saml.sp;
+
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.TimeValue;
+import org.joda.time.Duration;
+import org.joda.time.ReadableDuration;
+import org.opensaml.saml.saml2.core.NameID;
+
+/**
+ * Defines default values for a service provider if they are not configured in {@link SamlServiceProviderDocument}
+ */
+public final class ServiceProviderDefaults {
+
+    public static final Setting<String> APPLICATION_NAME_SETTING
+        = Setting.simpleString("xpack.idp.privileges.application", Setting.Property.NodeScope);
```

Comment:
add under `xpack.idp.defaults` namespace too ?

Reply:
I intentionally didn't, because of https://github.com/elastic/elasticsearch/pull/53320

The application here will become the one and only application not the "default", so it seemed strange to put it under `defaults`.
That makes it a little weird to be on this class, but ...